R/doeAnalysis.R

```diff
@@ -833,6 +842,67 @@ get_levels <- function(var, num_levels, dataset) {
   plot$plotObject <- p
 }
 
+.doeAnalysisPlotEffectNormalDistribution <- function(jaspResults, options, blocks, covariates, ready) {
+  if (!is.null(jaspResults[["normalEffectsPlot"]]) || !options[["normalEffectsPlot"]]) {
+    return()
+  }
+  plot <- createJaspPlot(title = gettext("Normal Plot of Standardized Effects"), width = 600, height = 600)
+  plot$dependOn(options = c("normalEffectsPlot", "tableAlias", .doeAnalysisBaseDependencies()))
+  plot$position <- 11
+  jaspResults[["normalEffectsPlot"]] <- plot
+  if (!ready || is.null(jaspResults[["doeResult"]]) || jaspResults$getError()) {
+    return()
+  }
+  result <- if (options[["codeFactors"]]) jaspResults[["doeResultCoded"]]$object[["regression"]] else jaspResults[["doeResult"]]$object[["regression"]]
+  fac <- if (options[["tableAlias"]]) result[["coefficients"]][["termsAliased"]][-1] else result[["coefficients"]][["terms"]][-1]
+  coefDf <- data.frame(result[["objectSummary"]]$coefficients)
+  tDf <- data.frame("tValue" = coefDf[["t.value"]],
+                    "terms" = result[["coefficients"]][["terms"]],
+                    "pValue" = result[["coefficients"]][["p"]])
+
+  # Do not include intercept, covariates and blocks in normal effects plot
+  tDf <- tDf[-1, ] # remove intercept
+  if (length(blocks) > 0 && !identical(blocks, "")) {
+    tDf <- tDf[!grepl(blocks, tDf$terms),]
+    fac <- if (options[["tableAlias"]]) fac[!grepl("BLK", fac)] else fac[!grepl(blocks, fac)]
+  }
+  if (length(covariates) > 0 && !identical(covariates, "")) {
+    tDf <- tDf[!tDf$terms %in% unlist(covariates), ] # remove the covariate(s)
+    fac <- if (options[["tableAlias"]]) fac[!grepl("COV", fac)] else fac[!fac %in% unlist(covariates)]
+  }
+
+  tDf$fac <- fac
+
+  # median rank order function
+  x <- tDf$tValue[order(tDf$tValue)]
+  n <- length(x)
+  i <- rank(x)
+  p <- (i - 0.3) / (n + 0.4)
+  tDf$percentile <- p[order(tDf$tValue)]
+
+  # statistical significance
+  tDf$significant <- ifelse(tDf$pValue < 0.05, "S", "N")
+  tDf$labelYPos <- pnorm(qnorm(tDf$percentile) + 0.2)
```

Comment:
do you have a reference for this? Stuff like `pnorm(qnorm(tDf$percentile) + 0.2)` looks a bit weird (why `+ 0.2`?). Also, you need to prefix with `stats::`, so `stats::pnorm(stats::qnorm(tDf$percentile) + 0.2)`

Reply:
If you mean the median rank function, yes, there are references for that (https://support.minitab.com/en-us/minitab/help-and-how-to/graphs/probability-plot/methods-and-formulas/method-of-obtaining-probability-plot-points/ ; http://engineeredsoftware.com/nasa/pe_median.htm). 

The `stats::pnorm(stats::qnorm(tDf$percentile) + 0.2)` is just some offset that seems to work reasonably well for placing the labels above the points. I found it easier/clearer to use a number on the percentage scale and then transform it than offsetting by an already transformed number. 

I have added a comment and the prefixes.